app/services/state_file/ty23_archiver_service.rb

```diff
@@ -0,0 +1,74 @@
+module StateFile
+  class Ty23ArchiverService
+    INTAKE_MAP = {
+      'az' => StateFileAzIntake,
+      'ny' => StateFileNyIntake,
+    }.freeze
+
+    attr_reader :state_code, :batch_size, :data_source, :tax_year, :current_batch, :cutoff
+
+    def initialize(state_code:, batch_size: 100, cutoff: '2024-06-01')
+      @state_code = state_code
+      @batch_size = batch_size
+      @data_source = INTAKE_MAP[state_code]
+      @tax_year = 2023
+      @cutoff = cutoff
+      @current_batch = nil
+      raise ArgumentError, "#{state_code} isn't an archivable state. Expected one of #{INTAKE_MAP.keys.join(', ')}" unless data_source
+    end
+
+    def find_archiveables
+      @current_batch = ActiveRecord::Base.connection.exec_query(query_archiveable)
+      Rails.logger.info("Found #{current_batch.count} #{data_source} intakes to archive.")
+    end
+
+    def archive_batch
+      archived_ids = []
+      current_batch.each do |record|
+        archive = StateFileArchivedIntake.new(record.without('source_intake_id'))
+        archive.submission_pdf.attach(data_source.find(record['source_intake_id']).submission_pdf.blob)
+        archive.save!
+        archived_ids << record['source_intake_id']
+      rescue StandardError => e
+        Rails.logger.warn("Caught exception #{e} for record #{record}. Continuing with batch")
+        next
+      end
+      Rails.logger.info("Archived #{archived_ids.count} #{data_source} intakes: [#{archived_ids.join(', ')}]")
+      @current_batch = nil # reset the batch
```

Comment:
why is the `@current_batch` set to nil for this method?

Reply:
ah, i think it is to exit the `while` loop in the rake task.

Reply:
setting it to `nil` to represent that the batch has been processed; those intakes are no longer eligible for archiving because they've been archived :)